src/modules/dashboard/components/CreateColonyWizard/TokenSelector.tsx

```diff
@@ -93,16 +102,20 @@ const TokenSelector = ({
   }, [apolloClient, tokenAddress]);
 
   const [isLoading, setLoading] = useState(false);
+  const [hasError, setHasError] = useState(false);
 
   const handleGetTokenSuccess = useCallback(
     (token: OneToken) => {
-      const { name, symbol } = token;
+      const { name, symbol } = token || {};
       setLoading(false);
+
       if (!name || !symbol) {
-        onTokenSelect(null);
+        onTokenSelect(false, null);
+        setHasError(true);
+        onTokenSelectError(true);
```

Comment:
I think you can eliminate the `hasError` state in this component and just pass it as a prop.
Please, check React best practices/pros & cons of doing so.

Reply:
I have refactored the state management and moved the states of `Error` and `isCheckingAddress` to the parent.